install-deps

```diff
@@ -144,6 +144,14 @@ elif [[ "$(uname)" == 'Linux' ]]; then
             sudo add-apt-repository -y ppa:chris-lea/zeromq
             sudo add-apt-repository -y ppa:chris-lea/node.js
         fi
+
+        gcc_major_version=$(gcc --version | grep ^gcc | awk '{print $4}' | \
```

Comment:
I think you'd want to first check the value of the variable $CC (or $CXX), as they might get precedence during build.

That's also an easy way to fix the issue of CUDA only supporting gcc-4.9 and below. (which is the reason for your install errors)